src/rgw/rgw_data_sync.cc

```diff
@@ -2530,11 +2534,12 @@ int RGWBucketShardIncrementalSyncCR::operate()
           marker_tracker.try_update_high_marker(cur_id, 0, entry->timestamp);
           continue;
         }
-        if (make_pair<>(entry->timestamp, entry->op) != squash_map[make_pair(entry->object, entry->instance)]) {
+        if (entry->instance.empty() &&
+            make_pair<>(entry->timestamp, entry->op) != squash_map[make_pair(entry->object, entry->instance)]) {
```

Comment:
@cbodley I'm not extremely happy about this fix, would rather understand the exact root cause before. The mtime issue was a problem, but I think there's another issue coming into play (I think it's del squashing unlink_instance), but still need to verify.